Clamp the currentCostume number when loading a project

kchadha opened the pull request:
### Resolves

Resolves an issue reported by sentry `cannot find property bitmapResolution of undefined`. This was causing a BSOD in at least the following two projects:

This sb2 project:
65811674

This sb3 project: 
277128060

### Proposed Changes

When loading a project, clamp the currentCostume number to an appropriate value (an index available in the actual list of the target's costumes).

### Reason for Changes

Fixes the projects so that they can load.

Filed a separate issue (#1901) for trying to figure out how the project got saved in this state in the first place...

### Test Coverage

Tested with existing projects reporting this error on sentry.

thisandagain commented on src/serialization/sb2.js:
```diff
@@ -812,7 +816,9 @@ const specMapBlock = function (block) {
     if (!mapped) {
-        log.warn(`Couldn't find SB2 block: ${opcode}`);
+        log.warn(`Couldn't find SB2 block: ${opcode}, trying as custom procedure call`);
+        // block = ['call', block];
```
Looks like this may be some cruft that was left in.

kchadha replied:
Done.

thisandagain reviewed (changes requested):
Looks good! One minor nit.

kchadha committed:
When loading a project, clamp the currentCostume number to an appropriate value.

kchadha committed:
Log a warning for currentCostume out of range when incorrect project is being serialized.

kchadha commented:
@thisandagain, I removed the cruft and also added some new code: a log warning and a patch for *serialization* so that we don't continue to contribute to saving broken projects. There is a new TODO comment that references #1901.
